WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewController.swift

```diff
@@ -107,29 +107,37 @@ private extension ShippingLabelFormViewController {
         cell.configure(state: row.cellState,
                        icon: .shippingImage,
                        title: Localization.shipFromCellTitle,
-                       body: "To be implemented",
+                       body: viewModel.originAddress?.fullNameWithCompanyAndAddress,
                        buttonTitle: Localization.continueButtonInCells) { [weak self] in
             guard let self = self else { return }
             let shippingLabelAddress = self.viewModel.originAddress
-            let shippingAddressValidationVC = ShippingLabelAddressFormViewController(siteID: self.viewModel.siteID,
-                                                                                     type: .origin,
-                                                                                     address: shippingLabelAddress)
-            self.navigationController?.pushViewController(shippingAddressValidationVC, animated: true)
+            let shippingAddressVC = ShippingLabelAddressFormViewController(siteID: self.viewModel.siteID,
+                                                                           type: .origin,
+                                                                           address: shippingLabelAddress,
+                                                                           completion: { [weak self] (shippingLabelAddress) in
+                                                                            guard let self = self else { return }
+                                                                            self.viewModel.handleOriginAddressValueChanges(address: shippingLabelAddress)
+                                                                           })
```

Comment:
I hate this Xcode indentation! 🤪

Reply:
Same, at the same time there is no other way to stay in 160 characters :/